src/Microsoft.AspNetCore.SignalR.Client/InvocationSubject.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    internal class InvocationSubject : IObservable<object>
+    {
+        private readonly object _lock = new object();
+        private IList<IObserver<object>> _observers = new List<IObserver<object>>();
+        private Exception _error = null;
+        private bool _completed = false;
+
+        public IDisposable Subscribe(IObserver<object> observer)
+        {
+            lock (_lock)
+            {
+                if (_completed)
+                {
+                    observer.OnCompleted();
+                }
+                else if (_error != null)
+                {
+                    observer.OnError(_error);
+                }
+                else
+                {
+                    _observers.Add(observer);
+                }
+                return new Subscription(this, observer);
```

Comment:
Should probably return an empty disposable or something in the cases where `observer` isn't added to the list